spark/BidResponseJob.scala

```diff
@@ -0,0 +1,191 @@
+package com.liveintent.dwh.logs.jobs
+
+import com.liveintent.dwh.dmp.models._
+import com.liveintent.dwh.dmp.sources.BidderGroupInfoSource
+import com.liveintent.dwh.spark.util.{SparkArgs, SparkDwhJob}
+import com.liveintent.dwh.userver.sources.ImpressionSource
+import com.liveintent.util.logparser.LogParser
+import com.twitter.algebird.Moments
+import org.apache.spark.rdd.RDD
+
+
+class BidResponsesJob(override val args: SparkArgs) extends SparkDwhJob(args) {
+
+  import BidResponsesJob._
+
+  private lazy val impressions = ImpressionSource()
+    .map{ _.sanitizeToPositive }
+    .filter { i => i.isFirstImpression && i.demandType == Some("default") }
+    .groupBy { e => (e.decisionId.getOrElse(""), e.redecisionAttempt.getOrElse(-1)) }
+    .mapValues(_.take(1))
+    .map( _._2 )
```

Comment:
If the purpose is to check their ability to understand the magic in the companion object, I think these "business logic"-lines can be left out. I also think a comment on what an impression is and what the job is supposed to do as such would be helpful